src/userpreferences/repository.ts

```diff
@@ -0,0 +1,23 @@
+export interface UserPreferencesDataRepository {
+    /**
+     * Get user preferences
+     *
+     * @param key - name of the key containing the data
+     * @returns the data
+     */
+    get<T = any>(key?: string): T;
+
+    /**
+     * Set user data
+     *
+     * @param key - name of the key for the data
+     * @param data - the data to save, will delete the data if undefined
+     */
+    set<T = any>(key: string, data?: T): void;
```

Comment:
Shouldn't this return a promise?
```suggestion
    set<T = any>(key: string, data?: T): Promise<void>;
```

Reply:
I guess that depends on whether we want to be able to tell the consumer that something went wrong. If we consider it a fire-and-forget type action, we don't need a promise. But, since certain changes may require the user to either reload the app, or log out and log back in again, we probably want to know when the save is done, so the page can safely be reloaded…

The consumer can always just ignore the promise if they don't care, so it seems like a good idea to return at promise 👍

Reply:
Fixed by Bef and verified by me, so I'm resolving this thread 🙂